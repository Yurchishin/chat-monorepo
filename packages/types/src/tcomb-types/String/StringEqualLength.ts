import t from 'tcomb-validation'
import R from '@chat/ramda'

export const StringEqualLength = (value: number) => t.refinement(
    t.String,
    R.equalsLength(value),
    'StringEqualLength',
)
