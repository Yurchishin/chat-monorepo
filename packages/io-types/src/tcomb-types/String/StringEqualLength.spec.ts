import {StringEqualLength} from './StringEqualLength'
import {expectTypeSuccess, expectTypeFailure} from "../../tests"

describe('StringEqualLength(5)', () => {
	const eqLength5 = StringEqualLength(5)

	expectTypeSuccess(eqLength5, '55555')
	expectTypeFailure(eqLength5, '666666')
	expectTypeFailure(eqLength5, '4444')
	expectTypeFailure(eqLength5, 1)
})
