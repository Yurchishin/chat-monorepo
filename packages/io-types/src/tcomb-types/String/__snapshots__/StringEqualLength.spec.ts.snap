// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StringEqualLength(5) expectTypeFailure(1): TYPE_FAILURE::StringEqualLength(1) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": 1,
      "expected": [Function],
      "message": "Invalid value 1 supplied to String",
      "path": Array [],
    },
  ],
  "value": 1,
}
`;

exports[`StringEqualLength(5) expectTypeFailure(4444): TYPE_FAILURE::StringEqualLength(4444) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": "4444",
      "expected": [Function],
      "message": "Invalid value \\"4444\\" supplied to StringEqualLength",
      "path": Array [],
    },
  ],
  "value": "4444",
}
`;

exports[`StringEqualLength(5) expectTypeFailure(666666): TYPE_FAILURE::StringEqualLength(666666) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": "666666",
      "expected": [Function],
      "message": "Invalid value \\"666666\\" supplied to StringEqualLength",
      "path": Array [],
    },
  ],
  "value": "666666",
}
`;
